src/Experiment/src/MethodBuilder.cs

```diff
@@ -0,0 +1,191 @@
+            _parameters = mthdParams;
+            DeclaringType = declaringType;
+            Module = declaringType.Module;
+        }
+
+        internal MethodDefinitionHandle AppendMetadata(MetadataBuilder _metadata)
+        {
+            // Encoding return type and parameters.
+            var methodSignature = new BlobBuilder();
+            ParametersEncoder _parEncoder;
+            ReturnTypeEncoder _retEncoder;
+            new BlobEncoder(methodSignature).
+                MethodSignature(). // Need to add in calling conventions. 
+                Parameters((_parameters==null)? 0 :_parameters.Length, out _retEncoder, out _parEncoder);
+
+            if (_returnType != null && _returnType!=typeof(void)) 
+            {
+                MapReflectionTypeToSignatureType(_retEncoder.Type(), _returnType);
+            }
+            else // If null mark ReturnTypeEncoder as void
+            {
+                _retEncoder.Void();
+            }
+            if (_parameters != null) // If parameters null, just keep empty ParametersEncoder empty
+            {
+                foreach (var parameter in _parameters)
+                {
+                    MapReflectionTypeToSignatureType(_parEncoder.AddParameter().Type(), parameter);
+                }
+            }
+
+            // Create the method
+            _handle = _metadata.AddMethodDefinition(
+                Attributes,
+                MethodImplAttributes.IL,
+               _metadata.GetOrAddString(Name),
+               _metadata.GetOrAddBlob(methodSignature),
+               -1, //No body with interfaces
+               parameterList: default);
+            return _handle;
+        }
+
+        private static void MapReflectionTypeToSignatureType(SignatureTypeEncoder signature, Type type)
+        {
+            //We need to translate from Reflection.Type to SignatureTypeEncoder. Only most common types supported for the prototype.
+            switch (type.FullName)
+            {
+                case "System.Boolean":
+                    signature.Boolean();
+                    break;
+                case "System.Byte":
+                    signature.Byte();
+                    break;
+                case "System.Char":
+                    signature.Char();
+                    break;
+                case "System.Double":
+                    signature.Double();
+                    break;
+                case "System.Int32":
+                    signature.Int32();
+                    break;
+                case "System.Int64":
+                    signature.Int64();
+                    break;
+                case "System.Object":
+                    signature.Object();
+                    break;
+                case "System.String":
+                    signature.String();
+                    break;
+                default: throw new NotImplementedException("This parameter type is not yet supported");
+            }
+        }
```

Comment:
```suggestion
        => type.FullName switch
        {
            "System.Boolean" => signature.Boolean(),
            "System.Byte"       => singature.Byte(),
            //...
            _                           => throw new NotImplementedException("This parameter type is not yet supported");
        };
```
NIT: use the new Switch syntax if you want to.